astropy/io/fits/hdu/image.py

```diff
@@ -187,9 +187,20 @@ def section(self):
         return the appropriate slice of the data, and loads *only* that section
         into memory.
 
-        Sections are mostly obsoleted by memmap support, but should still be
-        used to deal with very large scaled images.  See the
-        :ref:`astropy:data-sections` section of the Astropy documentation for
+        Sections are useful for retrieving a small subset of data from a remote
+        file that has been opened with the ``use_fsspec=True`` parameter.
+        For example, you can use this feature to download a small cutout from
+        a large FITS image hosted in the Amazon S3 cloud (see the
+        :ref:`astropy:fits-cloud-files` section of the Astropy
+        documentation for more details.)
+
+        For local files, sections are mostly obsoleted by memmap support, but
+        should still be used to deal with very large scaled images.
+
+        Note that sections cannot currently be written to.  Moreover, any
+        in-memory updates to the image's ``.data`` property may not be
+        reflected in the slices obtained via ``.section``. See the
+        :ref:`astropy:data-sections` section of the documentation for
```

Comment:
I expanded the docstring of `ImageHDU.section` here to explain that this feature is useful when working with remote files, for which memmap support is not available.